chia/util/streamable.py

```diff
@@ -123,26 +151,26 @@ def convert_optional(convert_func: ConvertFunctionType, item: Any) -> Any:
 
 
 def convert_tuple(convert_funcs: List[ConvertFunctionType], items: Collection[Any]) -> Tuple[Any, ...]:
-    if len(items) != len(convert_funcs):
-        raise ValueError(f"Invalid size. Expected: {len(convert_funcs)}, got: {len(items)}")
     if not isinstance(items, (list, tuple)):
-        raise TypeError(f"expected: tuple or list, actual: {type(items).__name__}")
+        raise InvalidTypeError(tuple, type(items))
+    if len(items) != len(convert_funcs):
+        raise InvalidSizeError(len(convert_funcs), len(items))
     return tuple(convert_func(item) for convert_func, item in zip(convert_funcs, items))
 
 
 def convert_list(convert_func: ConvertFunctionType, items: List[Any]) -> List[Any]:
     if not isinstance(items, list):
-        raise TypeError(f"expected: list, actual: {type(items).__name__}")
+        raise InvalidTypeError(list, type(items))
     return [convert_func(item) for item in items]
 
 
 def convert_hex_string(item: str) -> bytes:
     if not isinstance(item, str):
-        raise TypeError(f"expected: hex-string, actual: {type(item).__name__}")
+        raise InvalidTypeError(str, type(item))
     try:
         return hexstr_to_bytes(item)
     except Exception as e:
-        raise TypeError(f"Can't convert the string {item!r} to bytes: {e}") from e
+        raise ConversionError(item, bytes, str(e)) from e
```

Comment:
Maybe just pass in `e` itself?  Then you can use `traceback` to include the original exception type which is often nice to see along with the message.  Maybe something like `''.join(traceback.format_exception_only(type(x), value=x)).strip()`.